### ISSUE TITLE
Borderless Fullscreen to prevent flicker when switching apps

### ISSUE THREAD

**Issue by FrostKiwi** [alacritty/alacritty#6124]
On my Windows 10 laptop, switching to fullscreen, when Alacritty is the sole fullscreen app, causes Windows to make the screen black for 3 seconds and all input is paused. I presume, this is due to "Exclusive Fullscreen" kicking in.
It is highly annoying, as it forces a pause on all input for 3 seconds, just because you alt-tabbed. This does not happen with a maximized alacritty window, but I would prefer using Alacritty without the window border. I believe a Borderless-Windowed mode would solve this problem.
I see MacOS having "Simplified Fullscreen" listed, but Windows has no such option.

### System

OS: Windows 10
Version: `alacritty 0.10.1 (2844606)`

**Comment by chrisduerr** [comment]
Can't you just disable Alacritty's decorations?

**Comment by FrostKiwi** [comment]
> Can't you just disable Alacritty's decorations?

Good point!
Strangely enough, I can't get alacrity to maximize though. Setting startup mode to maximized just produces a small rectangle in the top left. Setting Fullscreen + Disabled decorations gives the same result of 3 seconds black screen when alt-tabbing.
![image](https://user-images.githubusercontent.com/60887273/173259546-d1ecfd8f-281f-4e04-9820-75b6a9921dd2.png)

**Comment by FrostKiwi** [comment]
> Are you saying that you can't maximize Alacritty at all, with or without decorations?

No, with decorations it's fine. But without it just becomes this small square and startup_mode: Maximized is not being honored. It just stays that small square and Key-Combos like [Win] + [ArrowUp] are ignored.

**Comment by chrisduerr** [comment]
Huh, that' sounds like a bug. Not sure if in Alacritty or Windows.

I probably accept adding a borderless fullscreen mode for windows/linux, but honestly I have no interest implementing it myself since there's really no reason why this shouldn't be configurable in the OS.

**Comment by FrostKiwi** [comment]
> there's really no reason why this shouldn't be configurable in the OS.

OMG YES. The amount of borderless window patches that exist for video games is absurd.
Even so, I want this issue to exist so others may categorize this behavior of the come across it...

**Comment by chrisduerr** [comment]
Disabling compositing for fullscreen windows should be a configurable option in any compositor tbh, with the option to disable it on a per-application basis. Compton didn't get much right, but at least it managed that much.

**Comment by FrostKiwi** [comment]
Ohh, I was referring to Windows...

**Comment by chrisduerr** [comment]
Which has a development budget that should blow `compton` out of the water...

**Comment by Deliganli** [comment]
what is described here exactly happens same in Windows 11 as well with alacritty 0.10.1
